torch/fx/graph_module.py

```diff
@@ -188,10 +174,11 @@ def __init__(self, root: Union[torch.nn.Module, Dict[str, Any]], graph: Graph, c
             # ``foo.bar.baz``
             targets_to_copy.sort(key=lambda t: t.count('.'))
             for target_to_copy in targets_to_copy:
-                _assign_attr(root[target_to_copy], self, target_to_copy)
+                self.insert_submodule(target_to_copy, root[target_to_copy])
         else:
             raise RuntimeError('Unsupported type ' + str(root) + ' passed for root!')
         self.graph = graph
+        graph._gm = self
```

Comment:
I don't think it makes sense for `GraphModule` to be assigning private attributes on `Graph`. I have a few other questions as well:

1) What happens if `Graph` is already owned by another `GraphModule`?
2) Going off 1, should we make it so that `_gm` is `None` in that situation? I guess the semantics could be "_gm contains the single owning `GraphModule` for this graph, otherwise (0 or 2+ owners) it is `None)"

Reply:
Also I think line 181 is unnecessary since `self.graph = graph` will call the `graph` setter below